File: public/dsa/linked_lists/linked_list_to_array/approach/linked_list_to_array.cpp

#include <bits/stdc++.h>
using namespace std;

// defs-start
class ListNode {
   public:
    int data;
    ListNode* next;
    ListNode(int data, ListNode* next = nullptr) : data(data), next(next) {}
};

// defs-end

vector<int> linkedListToArray(ListNode* head) {
    // add your logic here
    vector<int> arr;
    auto temp = head;
    while (temp) {
        arr.push_back(temp->data);
        temp = temp->next;
    }
    return arr;
}

// tests-start
int main() {
    ListNode* head = new ListNode(1);
    head->next = new ListNode(3);
    head->next->next = new ListNode(4);
    head->next->next->next = new ListNode(7);

    auto arr = linkedListToArray(head);

    for (int v : arr) {
        cout << v << " ";
    }

    return 0;
}
// tests-end

/*output
1 3 4 7
*/

File: public/dsa/linked_lists/linked_list_to_array/approach/linked_list_to_array.py
from typing import List


# defs-start
class ListNode:
    def __init__(self, data=0, next=None):
        self.data = data
        self.next = next
# defs-end


class Solution:
    def linkedListToArray(self, head: ListNode) -> List[int]:
        arr = []
        temp = head
        while temp:
            arr.append(temp.data)
            temp = temp.next
        return arr


# tests-start
head = ListNode(1, ListNode(3, ListNode(4, ListNode(7))))
sol = Solution()
print(sol.linkedListToArray(head))
# tests-end

'''output
[1, 3, 4, 7]
'''
